internal/resources/iampolicy/resource_iam_policy.go

```diff
@@ -0,0 +1,503 @@
+			}
+
+			return diag.FromErr(errors.Wrapf(err, "unable to get Role Bindings for cluster group"))
+		}
+
+		policyList = resp.PolicyList
+	case cluster:
+		fn, _ := fnData.(*clustermodel.VmwareTanzuManageV1alpha1ClusterFullName)
+
+		resp, err := config.TMCConnection.ClusterIAMResourceService.ManageV1alpha1ClusterIAMPolicyGet(fn)
+		if err != nil || resp == nil {
+			if clienterrors.IsNotFoundError(err) {
+				d.SetId("")
+				return diags
+			}
+
+			return diag.FromErr(errors.Wrapf(err, "unable to get Role Bindings for cluster"))
+		}
+
+		policyList = resp.PolicyList
+	case workspace:
+		fn, _ := fnData.(*workspacemodel.VmwareTanzuManageV1alpha1WorkspaceFullName)
+
+		resp, err := config.TMCConnection.WorkspaceIAMResourceService.ManageV1alpha1WorkspaceIAMPolicyGet(fn)
+		if err != nil || resp == nil {
+			if clienterrors.IsNotFoundError(err) {
+				d.SetId("")
+				return diags
+			}
+
+			return diag.FromErr(errors.Wrapf(err, "unable to get Role Bindings for workspace"))
+		}
+
+		policyList = resp.PolicyList
+	case namespace:
+		fn, _ := fnData.(*namespacemodel.VmwareTanzuManageV1alpha1ClusterNamespaceFullName)
+
+		resp, err := config.TMCConnection.NamespaceIAMResourceService.ManageV1alpha1ClusterNamespaceIAMPolicyGet(fn)
+		if err != nil || resp == nil {
+			if clienterrors.IsNotFoundError(err) {
+				d.SetId("")
+				return diags
+			}
+
+			return diag.FromErr(errors.Wrapf(err, "unable to get Role Bindings for namespace"))
+		}
+
+		policyList = resp.PolicyList
+	}
+
+	d.SetId(d.State().ID)
+
+	if err := d.Set(common.MetaKey, common.FlattenMeta(meta)); err != nil {
+		return diag.FromErr(err)
+	}
+
+	// iterate over the policy lists got in response and store in rbServerList with same UIDs
+	for _, policy := range policyList {
+		if policy.Meta.UID == d.State().ID {
+			rbServerList = append(rbServerList, policy.RoleBindings...)
+		}
+	}
+
+	// compare the role bindings got from state and server and store the intersection of the two.
+	for _, stateRB := range rbStateList {
+		for _, serverRB := range rbServerList {
+			if stateRB.Role == serverRB.Role {
+				calRBList = append(
+					calRBList,
+					&iammodel.VmwareTanzuCoreV1alpha1PolicyRoleBinding{
+						Role:     stateRB.Role,
+						Subjects: getIntersectionOfSubs(stateRB.Subjects, serverRB.Subjects),
+					})
+			}
+		}
+	}
+
+	if err := d.Set(roleBindingsKey, flattenRoleBindingList(calRBList)); err != nil {
+		return diag.FromErr(err)
+	}
+
+	return diags
+}
+
+func resourceIAMPolicyInPlaceUpdate(ctx context.Context, d *schema.ResourceData, m interface{}) (diags diag.Diagnostics) {
+	config := m.(authctx.TanzuContext)
+
+	var UID string
+
+	s, err := getScopeType(d)
+	if err != nil {
+		return diag.FromErr(errors.Wrap(err, "unable to get Role Binding"))
+	}
+
+	fnData := constructScopeFullName(s, d)
+	if fnData == nil {
+		return diag.FromErr(errors.Wrap(err, "unable to get Role Binding, no full name found"))
+	}
+
+	oldValues, newValues := d.GetChange(roleBindingsKey)
```

Comment:
can you explain this, 
oldValues returned by GetChange will always have deleted ones ?